tracer/src/Datadog.Trace/Propagators/ParseUtility.cs

```diff
@@ -27,19 +27,17 @@ public static ulong ParseFromHexOrDefault(ReadOnlySpan<char> value)
 
             return default;
         }
-#else
+#endif
+
         public static ulong ParseFromHexOrDefault(string value)
         {
-            try
-            {
-                return Convert.ToUInt64(value, 16);
-            }
-            catch
+            if (ulong.TryParse(value, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var result))
```

Comment:
What was the reason for this change? I wonder why it wasn't used in the first place 🤔 Also, any reason for not using the `#else`? We should always prefer that AFAIK, so using the `#else` would ensure the correct overload is always used?

Reply:
This was a little unrelated cleanup, as is my wont. We don't need this change in this PR so I can remove them. I found this code when working with `W3CTraceContextPropagator`. There are two changes here:
- Overloads: Previously we defined only `ParseFromHexOrDefault(ReadOnlySpan<char>)` for .NET Core and only `ParseFromHexOrDefault(string)` for .NET Framework. I thought it made more sense to _always_ have `ParseFromHexOrDefault(string)` (which works in any runtime) by dropping the `#else`, and _add_ the `ReadOnlySpan` overload when targeting .NET Core.
- Parsing: The .NET Framework code used `Convert.ToUInt64(value, 16)` for some reason. I thought it made more sense to use the overloads of `ulong.TryParse()` in both cases for consistency.

Reply:
Sorry for nitpicking your cleanup 😂 

> I thought it made more sense to always have ParseFromHexOrDefault(string) (which works in any runtime) by dropping the #else, and add the ReadOnlySpan overload when targeting .NET Core.

I'm not sure it _does_ make sense though 😄 I'm not sure there's a reason when we should prefer the `string` based overload over the span based overload, especially as there's an implicit conversion between `string` and `ReadOnlySpan<char>` 🤔

> Parsing: The .NET Framework code used Convert.ToUInt64(value, 16) **for some reason**. 

My question is: what's the reason? 😉 Both methods were added at the same time, so there presumably _was_ a reason, and I'm just hesitant to change things without knowing what it was. Chesterton's fence, and all 😉

Reply:
I'll revert change to this file and add a comment about using different implementations due to perf differences to avoid someone (probably me) trying this again in the future.

Reply:
Done!